cmd/start.go

```diff
@@ -126,7 +127,7 @@ func start(cmd *cobra.Command) error {
 	}
 
 	initializeMetrics(ctx, clientConfig, netProvider, blockCounter)
-	initializeDiagnostics(ctx, clientConfig, netProvider, signing)
+	initializeDiagnostics(clientConfig, netProvider, signing, rootCmd.Version)
```

Comment:
So what is currently returned in the diagnostics call is e.g. `"version":"v1.3.1-5900-g7600556b1 (revision 7600556b1)`.
I'm not sure reading the value from the root command's Version attribute is the best way.
For the command we may build the version output in a different way, we could add some more information (like contracts addresses, etc.), and structure the printed string differently.

For diagnostics, we may want to have a different pattern, e.g. instead of having one property: `"version":"v1.3.1-5900-g7600556b1 (revision 7600556b1)`.
We may want to structure them to validate them easier automatically in the monitoring solutions or for rewards allocation:

```json
"client_info":
    {
        "chain_address": "0xaabeB99Ece50a1dbeE06217e31fc45Bf18D37eEC",
        "network_id": "16Uiu2HAkwHzFwbHjoWRT7qPfmD7MzobSYSe8579ud1Rhjidssna8",
        "version": "v1.3.1-5900-g7600556b1",
        "revision": "7600556b1"
    },
```

We may need more flexibility.

We expose the `main.version` variable and do:
```
initializeDiagnostics(clientConfig, netProvider, signing, main.Version)
```

Reply:
Ok, I've split the version string to `version` and `revision` [719fbbf](https://github.com/keep-network/keep-core/pull/3234/commits/719fbbfaf17197d64afa5f0aaaa353381230e3c2)
Now
```
"client_info":
    {
        "chain_address": "0xaabeB99Ece50a1dbeE06217e31fc45Bf18D37eEC",
        "network_id": "16Uiu2HAkwHzFwbHjoWRT7qPfmD7MzobSYSe8579ud1Rhjidssna8",
        "version": "v1.3.1-5900-g7600556b1",
        "revision": "7600556b1"
    }
```

Reply:
Revision `7600556b1` is the commit hash, right? And what is `5900` part?

Reply:
> Revision 7600556b1 is the commit hash, right?
Right

> And what is 5900 part?
It is the number of commits on top of the `v1.3.1`. Note that `v1.3.1` was created on 2020/10/20 so we have 5900 commits since then.

Reply:
@pdyraga note that this is how `git describe` resolves the version. If we run it on a HEAD that was a tagged commit it will resolve to that tag, and not add the number of commits.

Reply:
Is this piece of information something we want to have in the version then?

I am missing some context in how we resolve the version but my understanding of @nkuba's comment is that this will *not* be added to the client build and it's here only because @dimpar built the client locally on `main` branch and there was no tag on that commit. Is this correct?

Instead of this piece of information, I think we should include the build tag (stable, dev) so our version should be: - For mainnet: `v1.3.1-stable-7600556b1`
For testnet dev builds: `v1.3.1-dev-7600556b1`(during the work on v1.3.1, before v1.3.1 is tagged)

Reply:
This is out of the scope of this PR. We set the version when building the client, and we can set it to anything we want. Let's discuss it in another thread.

Reply:
OK, if there is no problem in setting the version number during the build, I am fine!

Reply:
Option 2

Reply:
I misunderstood the original Kuba's comment and didn't get the 'main.Version' part.

Sure, option 2 sounds good. We have to provide the full path to the `build.Version` & `build.Revision` in the Makefile, otherwise it won't work. Here's the change [868f332](https://github.com/keep-network/keep-core/pull/3234/commits/868f332c2571d4ddc4f5cd263ca48f50d6d6bda7)